src/Streamly/Coreutils/Touch.hs

```diff
@@ -31,18 +33,18 @@ data Touch = Touch
 defaultConfig :: Touch
 defaultConfig = Touch On On
 
-setDeRef :: Switch -> Touch -> Touch
-setDeRef opt cfg = cfg {deRef = opt}
+followLinks :: Switch -> Touch -> Touch
+followLinks opt cfg = cfg {deRef = opt}
 
-setCreateNew :: Switch -> Touch -> Touch
-setCreateNew opt cfg = cfg {createNew = opt}
+create :: Switch -> Touch -> Touch
+create opt cfg = cfg {createNew = opt}
 
 touch :: (Touch ->Touch) -> FilePath -> IO ()
 touch f path = do
     let opt = f defaultConfig
     when (createNew opt == On) $ do
         found <- test path isExisting
-        unless found $ writeFile path ""
+        unless found $ void $ openFile path WriteMode
```

Comment:
Instead of discarding the result we should close the Handle gracefully, something like: `openFile path WriteMode >>= hClose`